### PR TITLE
Various enhancements

### PR DESCRIPTION
**Pull request by jschueller**
@regislebrun @sofianehaddad ready to go now:
- documentation updates
- code cleanup
- fix for chaospy thats breaking a test on the ci
- #1384
- #1388 
- #1386
- Add Sample::getMarginal(Description)

### CODE REVIEW

**Comment by jschueller** [comment]
This one is ready now, and necessary to fix the CI

File: lib/src/Base/Stat/SampleImplementation.cxx [hunk 1]
```diff
@@ -2261,6 +2261,20 @@ SampleImplementation SampleImplementation::getMarginal(const Indices & indices)
   return marginalSample;
 }
 
+/* Get the marginal by identifiers */
+SampleImplementation SampleImplementation::getMarginal(const Description & description) const
+{
+  Indices indices;
+  for (UnsignedInteger i = 0; i < description.getSize(); ++ i)
+  {
+    const UnsignedInteger index = getDescription().find(description[i]);
```
Comment (regislebrun) [inline lib/src/Base/Stat/SampleImplementation.cxx]: So if the description contains duplicate labels, the first one is used. It should be mentioned in the documentation.

File: python/src/Sample_doc.i.in [hunk 2]
```diff
@@ -856,7 +856,7 @@ n : int
 
 Parameters
 ----------
-indices : int or sequence of int, :math:`0 \leq i < n`
+indices : int, sequence of int, :math:`0 \leq i < n` or sequence of str
```
Comment (regislebrun) [inline python/src/Sample_doc.i.in]: Mention whats append if there are duplicate labels either in the sample description or in the filter description.

Comment (jschueller) [reply python/src/Sample_doc.i.in]: ok

**Pull request merged**